src/custom/components/WalletModal/WalletModalMod.tsx

```diff
@@ -211,13 +212,22 @@ export default function WalletModal({
     }
 
     connector &&
-      activate(connector, undefined, true).catch((error) => {
-        if (error instanceof UnsupportedChainIdError) {
-          activate(connector) // a little janky...can't use setError because the connector isn't set
-        } else {
-          setPendingError(true)
-        }
-      })
+      activate(connector, undefined, true)
+        .catch((error) => {
+          if (error instanceof UnsupportedChainIdError) {
+            activate(connector) // a little janky...can't use setError because the connector isn't set
+          } else {
+            setPendingError(true)
+          }
+        })
+        .then(() => {
+          // manually set the WalletConnectConnector http.connection.url to currently connected network url
+          if (connector instanceof WalletConnectConnector) {
+            const { http, rpc, signer } = connector.walletConnectProvider
+            const chainId = (signer.connection as SignerConnection).chainId
+            http.connection.url = rpc.custom[chainId || 1]
```

Comment:
hacky, but if it works, im fine. Only thing, can you add a comment, and also if there's a fix for WC point it to the PR or issue of the bug

Last but not least, could you avoid using "magic numbers", in this case we have a Chain enum

Reply:
Updated, I suggest we make sure this works correctly in our app as a fix and then I could create PR for WC package also.